src/index.js

```diff
@@ -1,15 +1,44 @@
 #!/usr/bin/env node
+//@ts-check
 
-import { Package, Step } from "./declarations/types";
-import { execSync } from "child_process";
-import fs from "fs-extra";
-import path from "path";
-import prompts from "prompts";
-import validateName from "validate-npm-package-name";
+const { execSync } = require("child_process");
+const fs = require("fs-extra");
+const path = require("path");
+const prompts = require("prompts");
+const validateName = require("validate-npm-package-name");
 
-const getApplicationId = (token: string): string | null => {
+/**
+ * @typedef {() => void} StepAction
+ */
+
+/**
+ * @typedef {Object} Package
+ * @property {string} name
+ * @property {string} version
+ * @property {string} description
+ * @property {Record<string, string>} scripts
+ * @property {Record<string, string>} dependencies
+ * @property {Record<string, string>} devDependencies
+ */
+
+/**
+ * @typedef {Object & { isDry: boolean; }} Step
+ * @property {string} message
+ * @property {StepAction} action
```

Comment:
maybe we can clean this typedef up by actually including `ignoreDry` in each step object (which is currently not included in any of the objects except for the last step)
```
@typedef {Object} Step
@property {string} message
@property {boolean} ignoreDry
@property {StepAction} action
```

Reply:
It is an optional value and using `boolean | undefined` seems to still cause compiler errors.

Reply:
yeah i was suggesting we could add ignoreDry to all the step objects (`ignoreDry: false`) so that it's no longer optional and we can clean up the typedef that way